qiita_pet/handlers/qiita_redbiom.py

```diff
@@ -0,0 +1,118 @@
+from tornado.gen import coroutine, Task
+
+from qiita_core.util import execute_as_transaction
+
+from .base_handlers import BaseHandler
+import redbiom.summarize
+import redbiom.search
+import redbiom._requests
+import redbiom.util
+import redbiom.fetch
+
+
+class RedbiomPublicSearch(BaseHandler):
+    @execute_as_transaction
+    def _get_context(self, callback):
+        df = redbiom.summarize.contexts()
+        result = df.ContextName.values
+        callback(result)
+
+    @coroutine
+    @execute_as_transaction
+    def get(self, search):
+        contexts = yield Task(self._get_context)
+        self.render('redbiom.html', contexts=contexts)
+
+    @execute_as_transaction
+    def _redbiom_search(self, context, query, search_on, callback):
+        df = redbiom.summarize.contexts()
+        contexts = df.ContextName.values
+        if context not in contexts:
+            callback(([], 'The given context is not valid: %s - %s' % (
+                context, contexts)))
+        else:
+            if search_on == 'metadata':
+                samples = redbiom.search.metadata_full(query, categories=None)
+            elif search_on == 'observations':
+                # from_or_nargs first parameter is the file handler so it uses
+                # that as the query input. None basically will force to take
+                # the values from query
+                samples = [s.split('_', 1)[1]
```

Comment:
What this is doing is converting from a redbiom ID to an ambiguous ID, and it does not seem to be enforcing that the ID is unique through a `set` comprehension. You may want to use `redbiom.util.resolve_ambiguities` instead